app/components/billing/payment.js

```diff
@@ -53,6 +60,38 @@ export default Component.extend({
     }
   }).drop(),
 
+  // amount_off and price are in cents
+  discountedPrice: computed('coupon.{amountOff,percentOff}', 'selectedPlan.price', function () {
+    const price = Math.floor(this.selectedPlan.price / 100);
+    if (this.coupon && this.coupon.amountOff) {
+      const { amountOff } = this.coupon;
+      const discountedPrice = price - Math.floor(amountOff / 100);
+      return `$${discountedPrice}`;
+    } else if (this.coupon && this.coupon.percentageOff) {
+      const { percentageOff } = this.coupon;
+      const discountedPrice = price - (price * percentageOff) / 100;
+      return `$${discountedPrice.toFixed(2)}`;
+    } {
+      return `$${price}`;
+    }
+  }),
+
+  validateCoupon: task(function* () {
```

Comment:
As long as `coupon` seems to belong to `subscription` model, I'd suggest to move validation logic there too, what do you think?

Reply:
As long as `task`s expose state, the ones that depend on certain data (e.g. `couponId` in this case) should not reside on services, because it could cause unexpected behavior.